data/tools/unit_tree/helpers.py

```diff
@@ -40,11 +27,15 @@ def __init__(self, wesnoth_exe, userdir, datadir, magick_exe="convert"):
         self.images_by_ipath = {}
         self.binary_paths_per_addon = {}
         self.datadir = datadir
-        if not self.datadir:
-            self.datadir = get_datadir(wesnoth_exe)
         self.userdir = userdir
-        if not self.userdir:
-            self.userdir = get_userdir(wesnoth_exe)
+        self.hide_paths = [
+            os.path.join(self.userdir, "data", "add-ons"),
+            os.path.join(self.userdir, "data"),
+            os.path.join(self.userdir),
+            os.path.join(self.datadir, "data", "add-ons"),
+            os.path.join(self.datadir, "data"),
+            os.path.join(self.datadir)
```

Comment:
Are these paths expected to exist somewhere? Why was this changed from the paths used before?

Reply:
These are the root dirs that were trimmed in make_name. There shouldn't be functional change in that regard. (But make_name produces different output now anyway.)

Reply:
Oh, I see what you mean, maybe. The ``get_datadir()`` and ``get_userdir()`` functions are just fallbacks that are not needed, since this constructor is always called with those parameters.

Reply:
As far as I can tell previously the paths were `options.config_dir`, `options.data_dir` and `"data"`.

`os.path.join(self.datadir, "data", "add-ons")` shouldn't exist at all.

Reply:
Right... Might as well omit it. As for config_dir/data/add-ons, I am also trimming ``add-ons`` in an effort to have the shortest feasible paths (Windows has a nasty max path length of 260 chars.)

Reply:
Alright, sounds good.